Services/AvatarService/Sources/AvatarService.swift

```diff
@@ -0,0 +1,67 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import Foundation
+import Store
+import Primitives
+import UIKit
+
+public struct AvatarService: Sendable {
+    private let store: WalletStore
+    
+    public init(store: WalletStore) {
+        self.store = store
+    }
+    
+    // MARK: - Store
+
+    @MainActor
+    public func save(image: UIImage, walletId: String) throws {
+        guard let data = image.compress(.highest) else {
+            throw AnyError("Compression image failed")
+        }
+        
+        let path = try preparePath(for: walletId)
+        try data.write(to: path, options: .atomic)
+        try store.setWalletAvatar(walletId, url: path)
+    }
+    
+    public func save(url: URL, walletId: String) async throws {
```

Comment:
```suggestion
    public func save(url: URL, for walletId: String) async throws {
```
make it consistent with other methods